core/src/Slay/Prim.hs

```diff
@@ -82,6 +86,18 @@ data PrimCurve g =
       curveDirection :: g Direction
     }
 
+
+-- | circle adt
+data PrimCircle g = Circle
+    { circleColor :: g (Maybe Color)
+    , circleCenterX :: Double
+    , circleCenterY :: Double
+    , circleRadius :: Double
```

Comment:
I used Double because arc accepts params as that type. I can make it Natural but we will need to go from Natural -> Double in the RenderElement instance